public/app/features/dashboard/panel_editor/QueriesTab.tsx

```diff
@@ -74,8 +75,14 @@ export class QueriesTab extends PureComponent<Props, State> {
       next: (data: PanelData) => this.onPanelDataUpdate(data),
     });
 
-    const ds = await getDataSourceSrv().get(panel.datasource);
-    this.setState({ dataSource: ds });
+    try {
+      const ds = await getDataSourceSrv().get(panel.datasource);
+      this.setState({ dataSource: ds });
```

Comment:
Is a dataSourceItem already set for the default case?

Reply:
yes, it is set in the initial state. Great question!